### ISSUE TITLE
Need I get a new cluster centroids when testing?

### ISSUE THREAD

**Issue by RaymondLmx** [Nanne/pytorch-NetVlad#36]
```
            if not opt.resume: 
                if opt.mode.lower() == 'train':
                    initcache = join(opt.dataPath, 'centroids', opt.arch + '_' + train_set.dataset + '_' + str(opt.num_clusters) +'_desc_cen.hdf5')
                else:
                    initcache = join(opt.dataPath, 'centroids', opt.arch + '_' + whole_test_set.dataset + '_' + str(opt.num_clusters) +'_desc_cen.hdf5')
```
In this part of code, I can confirm that I need to use the specifc desc_cen when testing, while the code just get cluster for whole_train_set. 
```
    elif opt.mode.lower() == 'cluster':
        print('===> Calculating descriptors and clusters')
        get_clusters(whole_train_set)
```
Does this mean when I want the desc_cen for test_test, I need to change the whole_train_set here to whole_test_set?

**Comment by Nanne** [comment]
As the first line shows, if you're not resuming from a trained model, then you will need to calculate cluster centroids. If you're resuming from a trained model then you shouldn't need to calculate clusters.

In practice I don't see why you would ever need to calculate clusters on the test set. 

I would recommend to study the NetVlad paper, rather than the code, in detail to get a better grip of what is happening and why.
